modules/acme/foobar/hooks/entrypointhook.go

```diff
@@ -18,10 +19,15 @@ type EntrypointHook struct {
 
 func (h EntrypointHook) Handle(
 	_ context.Context,
-	_ invocation.Context,
+	_ *invocation.ModuleContext,
```

Comment:
I think it's better to avoid passing `ModuleContext` to hooks as a pointer, otherwise, hooks can modify values not intended for modification.

Reply:
I think we can not worry about this as ModuleContext is intended only for module internal usage and they are responsible for any manipulations with them. Besides pointers will allow to use less memory allocations.